test/lib/sstable_test_env.hh

```diff
@@ -97,9 +98,9 @@ public:
         return reusable_sst(std::move(schema), dir, generation).then([] (auto ptr) { return make_ready_future<>(); });
     }
 
-    template <typename Func>
-    static inline auto do_with(Func&& func) {
-        return seastar::do_with(test_env(), [func = std::move(func)] (test_env& env) mutable {
+    template <typename Func, typename... Args>
+    static inline auto do_with(Func&& func, Args&&... args) {
```

Comment:
But why a template? nicer to have a config structure with optional everything in it.

This do_with() shouldn't be a template, it should accept a noncopyable_function instead (or std::move_only_function next year)

Reply:
Config is to put some "parameters" into the test_env, but this part is used to pass dependencies. The rest of the code passes dependencies via arguments.